### ISSUE TITLE
Local chain type

### ISSUE THREAD

**Issue by ebuchman** [informalsystems/ibc-rs#158]
We should add a new type of Chain type that is a local in-process chain. This chain will basically be our [implementation of the IBC handlers](https://github.com/informalsystems/ibc-rs/issues/119). It can also be used for in-process testing of the relayer.

A Chain is expected to be running a blockchain (generating at least light blocks ...) and to have some on-chain state that can be queried and updated by transactions. So we'll need:

- A way to produce a chain of real Tendermint light blocks (we should have all the tools for this now given the tendermint-rs model-based testing work to do this)
- An on chain kv-store. For now we should mock out proofs so we don't worry about checking them but soon we'll want to implement a basic merkle tree backing here, even a simple merkle tree that's recomputed every block. Eventually we'll use something like [merk](https://github.com/nomic-io/merk) or otherwise.
- A way to query the on chain kv-store (should roughly match the ABCI query interface)
- IBC handlers: ie. functions that take IBC msgs as input and executes them against the kv-store
- A way to send the msgs to the handlers (ie. "transactions")

So basically we need to mock out a Tendermint blockchain using light blocks and a simple state machine for the ibc handlers.

Note this wouldn't be using ABCI directly, since that would assume a running instance of Tendermint Core. So we don't need to hook up real ABCI queries or any kind of transaction stuff, just a simple interface for passing in the IBC msg types and for querying from the KV store. Ultimately though we'll want this application state/logic to be able to run on top of ABCI, so it's worth keeping in mind.
